contracts/feeds/PriceFeeds.sol

```diff
@@ -312,8 +312,9 @@ contract PriceFeeds is Constants, PausableGuardian {
         internal
         view
         returns (uint256 rate)
-    {
-        if (token != address(wethToken)) {
+    {   
+        // on ETH all pricefeeds are etherum denominated. on all other chains its USD denominated. so on ETH the price of 1 eth is 1 eth
+        if (getChainId() != 1 && token != address(wethToken)) {
```

Comment:
change this to be if (getChainId() != 1 || token != address(wethToken)) as this filters it correctly. If the chain id != 1 then it should always be getting the token price and if the chain id is 1 then it should be getting it only if the token is wethToken. As it is structured now, if the chain id is 1 then it always diverts to the else as one of the values in the AND operator is false.

Reply:
fixed